tasks/actorstate/task.go

```diff
@@ -69,6 +90,11 @@ func (t *Task) ProcessActors(ctx context.Context, current *types.TipSet, execute
 		return data, report, nil
 	}
 
+	if t.exceedActorChangeLimit(actors) {
```

Comment:
this and 0 length check should be right after the filtering (after line 80) so we don't need to allocate arrays for `data` and `errorsDetected`.

Reply:
nice catch!